crypto/evp/evp_enc.c

```diff
@@ -444,22 +457,81 @@ int EVP_CipherInit_ex2(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
 }
 
 int EVP_CipherInit(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
                    const unsigned char *key, const unsigned char *iv, int enc)
 {
     if (cipher != NULL)
         EVP_CIPHER_CTX_reset(ctx);
-    return evp_cipher_init_internal(ctx, cipher, NULL, key, iv, enc, NULL);
+    return evp_cipher_init_internal(ctx, cipher, NULL, key, iv, enc, 0, NULL);
 }
 
 int EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
                       ENGINE *impl, const unsigned char *key,
                       const unsigned char *iv, int enc)
 {
-    return evp_cipher_init_internal(ctx, cipher, impl, key, iv, enc, NULL);
+    return evp_cipher_init_internal(ctx, cipher, impl, key, iv, enc, 0, NULL);
+}
+
+static int evp_cipher_pipeline_init_internal(EVP_CIPHER_CTX *ctx,
+                                             const EVP_CIPHER *cipher,
+                                             const unsigned char *key,
+                                             size_t numpipes,
+                                             const unsigned char **iv, int enc,
+                                             const OSSL_PARAM params[])
+{
+    if (numpipes >= EVP_MAX_PIPES) {
+        ERR_raise(ERR_LIB_EVP, EVP_R_TOO_MANY_PIPES);
+        return 0;
+    }
+
+    if (!evp_cipher_init_internal(ctx, cipher, NULL, NULL, NULL, enc, 1,
+                                  params))
+        return 0;
+
+    if (enc) {
+        if (ctx->cipher->p_einit == NULL) {
+            ERR_raise(ERR_LIB_EVP, EVP_R_INITIALIZATION_ERROR);
+            return 0;
+        }
+
+        return ctx->cipher->p_einit(ctx->algctx,
+                                    key,
+                                    key == NULL ? 0
+                                                : EVP_CIPHER_CTX_get_key_length(ctx),
+                                    numpipes,
+                                    iv,
+                                    iv == NULL ? 0
+                                               : EVP_CIPHER_CTX_get_iv_length(ctx),
+                                    params);
+    }
+
+    if (ctx->cipher->p_dinit == NULL) {
+        ERR_raise(ERR_LIB_EVP, EVP_R_INITIALIZATION_ERROR);
+        return 0;
+    }
+
+    return ctx->cipher->p_dinit(ctx->algctx,
+                                key,
+                                key == NULL ? 0
+                                            : EVP_CIPHER_CTX_get_key_length(ctx),
+                                numpipes,
+                                iv,
+                                iv == NULL ? 0
+                                           : EVP_CIPHER_CTX_get_iv_length(ctx),
+                                params);
+}
+
+int EVP_CipherPipelineInit(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
+                           const unsigned char *key, size_t numpipes,
+                           const unsigned char **iv, int enc)
+{
+    if (cipher != NULL)
+        EVP_CIPHER_CTX_reset(ctx);
```

Comment:
Don't do this. Only the original Init functions did this. Subsequent "ex" variants haven't done so, so we shouldn't do it here.